modules/apps/frontend-taglib/frontend-taglib-clay-sample-web-test/src/testIntegration/java/com/liferay/frontend/taglib/clay/sample/web/test/ClaySamplePortletTest.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.taglib.clay.sample.web.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.layout.test.util.ContentLayoutTestUtil;
+import com.liferay.layout.test.util.LayoutTestUtil;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.Layout;
+import com.liferay.portal.kernel.service.ServiceContextThreadLocal;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.search.test.util.IndexerFixture;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LogEntry;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import java.util.List;
+import java.util.Locale;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Fortunato Maldonado
+ */
+@RunWith(Arquillian.class)
+public class ClaySamplePortletTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		_locale = _portal.getSiteDefaultLocale(_group);
+
+		_languageId = LocaleUtil.toLanguageId(_locale);
+
+		_layoutIndexerFixture = new IndexerFixture<>(Layout.class);
+
+		ServiceContextThreadLocal.pushServiceContext(
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+	}
+
+	@Test
+	public void testSampleClay() throws Exception {
+		ServiceContextThreadLocal.pushServiceContext(
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		Layout layout = LayoutTestUtil.addTypeContentLayout(_group);
+
+		Layout draftLayout = layout.fetchDraftLayout();
+
+		Assert.assertNotNull(draftLayout);
+
+		ContentLayoutTestUtil.addPortletToLayout(
+			layout, "com_liferay_clay_sample_web_portlet_ClaySamplePortlet");
```

Comment:
Hi @fortunatomaldonado ,

Here I think I should add the portlet to the draftLayout instead of the layout, so that it is copied to the published one when invoking ContentLayoutTestUtil.publishLayout(draftLayout, layout). In that way, the portlet will be on the layout when indexing;

As it is doing it here, the layout is always indexed empty because the portlet is removed when you publish it because is not on the draft layout. 

I hope that helps

Reply:
Thanks Lourdes, what you say makes sense to me. So, if I understood properly, flow should look something like this:

- create the layout
- grab the draft layout
- add the portlet to the draft layout
- capture the log
- publish the layout
- check log entries size

Is my understanding correct?

Reply:
Yes, exactly that. I think the only change needed is to use draftLayout instead of layout when invoking ContentLayoutTestUtil.addPortletToLayout

Reply:
Thank you @lfbesada! 
@dsanz 
I made the change and rebased master.